examples/vulkan/Vulkan.cpp

```diff
@@ -0,0 +1,2556 @@
+
+        return VK_FALSE;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+// VulkanExample class
+////////////////////////////////////////////////////////////
+class VulkanExample
+{
+public:
+    // Constructor
+    VulkanExample() :
+    window(sf::VideoMode(800, 600), "SFML window with Vulkan", sf::Style::Default),
+    vulkanAvailable(sf::Vulkan::isAvailable()),
+    maxFramesInFlight(2),
+    currentFrame(0),
+    swapchainOutOfDate(false),
+    instance(0),
+    debugReportCallback(0),
+    surface(0),
+    gpu(0),
+    queueFamilyIndex(-1),
+    device(0),
+    queue(0),
+    swapchainFormat(),
+    swapchainExtent(),
+    swapchain(0),
+    depthFormat(VK_FORMAT_UNDEFINED),
+    depthImage(0),
+    depthImageMemory(0),
+    depthImageView(0),
+    vertexShaderModule(0),
+    fragmentShaderModule(0),
+    descriptorSetLayout(0),
+    pipelineLayout(0),
+    renderPass(0),
+    graphicsPipeline(0),
+    commandPool(0),
+    vertexBuffer(0),
+    vertexBufferMemory(0),
+    indexBuffer(0),
+    indexBufferMemory(0),
+    textureImage(0),
+    textureImageMemory(0),
+    textureImageView(0),
+    textureSampler(0),
+    descriptorPool(0)
+    {
+        // Vulkan setup procedure
+        if (vulkanAvailable) setupInstance();
+        if (vulkanAvailable) setupDebugReportCallback();
+        if (vulkanAvailable) setupSurface();
+        if (vulkanAvailable) setupPhysicalDevice();
+        if (vulkanAvailable) setupLogicalDevice();
+        if (vulkanAvailable) setupSwapchain();
+        if (vulkanAvailable) setupSwapchainImages();
+        if (vulkanAvailable) setupShaders();
+        if (vulkanAvailable) setupRenderpass();
+        if (vulkanAvailable) setupDescriptorSetLayout();
+        if (vulkanAvailable) setupPipelineLayout();
+        if (vulkanAvailable) setupPipeline();
+        if (vulkanAvailable) setupCommandPool();
+        if (vulkanAvailable) setupVertexBuffer();
+        if (vulkanAvailable) setupIndexBuffer();
+        if (vulkanAvailable) setupUniformBuffers();
+        if (vulkanAvailable) setupDepthImage();
+        if (vulkanAvailable) setupDepthImageView();
+        if (vulkanAvailable) setupTextureImage();
+        if (vulkanAvailable) setupTextureImageView();
+        if (vulkanAvailable) setupTextureSampler();
+        if (vulkanAvailable) setupFramebuffers();
+        if (vulkanAvailable) setupDescriptorPool();
+        if (vulkanAvailable) setupDescriptorSets();
+        if (vulkanAvailable) setupCommandBuffers();
+        if (vulkanAvailable) setupDraw();
+        if (vulkanAvailable) setupSemaphores();
+        if (vulkanAvailable) setupFences();
+
+        // If something went wrong, notify the user by setting the window title
+        if (!vulkanAvailable)
+            window.setTitle("SFML window with Vulkan (Vulkan not available)");
+    }
+
+
+    // Destructor
+    ~VulkanExample()
+    {
+        // Wait until there are no pending frames
+        vkDeviceWaitIdle(device);
```

Comment:
Need to check for null device

Reply:
Not sure how much error handling one wants to add, but I guess if VK isn't available this may crash, @binary1248?